packages/language-server/src/logger.ts

```diff
@@ -1,44 +1,72 @@
 
-// TODO: This interface should be provided by the logging library.
-// Initially we are only providing a fairly simple API, additional variants can be added as needed...
-export interface ILogger {
-  /* eslint-disable @typescript-eslint/no-explicit-any -- meta is an object with string type key and any type value */
-  fatal: (msg: string, meta?: Record<string, any>) => void;
-  error: (msg: string, meta?: Record<string, any>) => void;
-  warn: (msg: string, meta?: Record<string, any>) => void;
-  info: (msg: string, meta?: Record<string, any>) => void;
-  debug: (msg: string, meta?: Record<string, any>) => void;
-  trace: (msg: string, meta?: Record<string, any>) => void;
-  /* eslint-enable @typescript-eslint/no-explicit-any -- meta is an object with string type key and any type value */
+// eslint-disable-next-line @typescript-eslint/no-var-requires -- Using `require` for .json file as this gets bundled with webpack correctly.
+const meta = require("../../package.json");
+
+export type ILogger = Omit<IChildLogger, "getChildLogger">;
+
+let logLevel: LogLevel = "error";
+/**
+ * We are using the VSCode Logging library right now as it is:
+ * 1. The only one of our logging libraries available on npmjs.com (currently).
+ * 2. Supports console logging which would be re-directed to the VSCode extension's output Channel
+ *    - Assuming this LSP server processes was spawned from the VSCode Extension
+ */
+const loggerImpl: IVSCodeExtLogger = getExtensionLogger({
+  extName: meta.name,
+  level: logLevel,
+  logConsole: true,
+});
+
+function buildSafeLoggerMethod(
+  methodName: LogLevel
+): (msg: string, ...args: unknown[]) => void {
+  return function (msg: string, ...args: unknown[]): void {
+    const safeArgs = map(args, removePossibleUserInformation);
+    loggerImpl[methodName].apply(loggerImpl, [msg, ...safeArgs]);
+  };
 }
 
-const NOOP_LOGGER: ILogger = {
-  /* eslint-disable @typescript-eslint/no-empty-function, @typescript-eslint/no-unused-vars, @typescript-eslint/no-explicit-any -- NOOP */
-  fatal: (msg: string, meta?: Record<string, any>) => {},
-  error: (msg: string, meta?: Record<string, any>) => {},
-  warn: (msg: string, meta?: Record<string, any>) => {},
-  info: (msg: string, meta?: Record<string, any>) => {},
-  debug: (msg: string, meta?: Record<string, any>) => {},
-  trace: (msg: string, meta?: Record<string, any>) => {},
-  /* eslint-enable @typescript-eslint/no-empty-function, @typescript-eslint/no-unused-vars, @typescript-eslint/no-explicit-any -- NOOP */
+const loggerWrapper: ILogger = {
+  fatal: buildSafeLoggerMethod("fatal"),
+  error: buildSafeLoggerMethod("error"),
+  warn: buildSafeLoggerMethod("warn"),
+  info: buildSafeLoggerMethod("info"),
+  debug: buildSafeLoggerMethod("debug"),
+  trace: buildSafeLoggerMethod("trace"),
 };
 
-let logger = NOOP_LOGGER;
 export function getLogger(): ILogger {
-  return NOOP_LOGGER;
+  return loggerWrapper;
+}
+
+const possibleSensitiveProps = { uri: true };
```

Comment:
Why is this an object instead of string array like before? You only use its keys

Reply:
not sure :)